### ISSUE TITLE
Training Loss not descending

### ISSUE THREAD

**Issue by Jaredanwolfgang** [pengzhangzhi/Open-dLLM#20]
Hi! 

I am trying to pre-train the Open-dCoder based on fine-code dataset and I run the train script with the following parameters:

```shell
data_parallel_mode: fsdp1
ulysses_parallel_size: 1
global_batch_size: 128
micro_batch_size: 8
lr: 3.0e-4
lr_decay_style: cosine
lr_warmup_ratio: 0.01
lr_decay_ratio: 1.0  # Use full training for decay
lr_min: 3e-6        # Minimum LR
```

Other params are using the default in the `qwen2_5_coder_500M.yaml` file. In the issue #8 it is said we ought to observe a large loss at the beginning of the training and a quick drop to loss around 2~3. This is also demonstrated in the [website](https://oval-shell-31c.notion.site/Open-dLLM-Open-Diffusion-Large-Language-Model-25e03bf6136480b7a4ebe3d53be9f68a). However, during my training, I haven't observed this phenomenon and the loss starts at around 2 at the beginning. 

<img width="362" height="315" alt="Image" src="https://github.com/user-attachments/assets/6b5beaa1-062f-484a-999d-f7c6f16e5f44" />

(The loss with large perturbation is using global batch size 128 and the low perturbation is using global batch size 512) I am not training a lot of steps but as demonstrated in the website the final loss should be fluctuating around the value of 2~3. Is this a normal phenomenon? How can I know whether the model is actually learning something? 

Thank you for your help!
